code/modules/mining/equipment_locker.dm

```diff
@@ -507,8 +507,8 @@
 	if(istype(proj_turf, /turf/simulated/mineral))
 		var/turf/simulated/mineral/M = proj_turf
 		playsound(src,'sound/weapons/resonator_blast.ogg',50,1)
-		M.gets_drilled()
 		spawn(5)
+			M.gets_drilled(creator)
```

Comment:
The other possibility sounds better to me, to be honest. I'd rather not rely on timing tricks for getting correct results unless we need to.